com.kms.cura.dal/src/com/kms/cura/dal/database/UserDatabaseHelper.java

```diff
@@ -46,39 +96,67 @@ protected void insertUser(UserEntity entity) throws SQLException, DALException,
 		}
 	}
 
-	@Override
-	protected UserEntity getEntityFromResultSet(ResultSet resultSet) throws SQLException, ClassNotFoundException {
-		return new UserEntity(resultSet.getString(UserColumn.ID.getColumnName()), null,
-				resultSet.getString(UserColumn.EMAIL.getColumnName()),
-				resultSet.getString(UserColumn.PASSWORD.getColumnName()));
+	protected PreparedStatement createPreparedStatement(Map<String, Object> valueMap, String databaseName,
+			String databaseMethod) throws SQLException {
+		StringBuilder insertString = new StringBuilder(databaseMethod + " " + databaseName + " (");
+		StringBuilder valueString = new StringBuilder(") VALUES (");
+		boolean isFirst = true;
+		for (Entry<String, Object> entry : valueMap.entrySet()) {
+			if (isFirst) {
+				isFirst = false;
+			} else {
+				insertString.append(",");
+				valueString.append(",");
+			}
+			insertString.append(entry.getKey());
+			valueString.append("?");
+		}
+		valueString.append(")");
+		String statementString = insertString.toString() + valueString.toString();
+		PreparedStatement stmt = con.prepareStatement(statementString);
+		setPreparedStatementValue(valueMap, stmt);
+		return stmt;
 	}
 
-	public UserEntity queryByEmail(String email) throws SQLException, ClassNotFoundException {
-		PreparedStatement stmt = null;
-		ResultSet rs = null;
+	protected void setPreparedStatementValue(Map<String, Object> valueMap, PreparedStatement stmt) throws SQLException {
+		int count = 1;
+		for (Entry<String, Object> valueEntry : valueMap.entrySet()) {
+			if (valueEntry.getValue() instanceof Integer) {
+				stmt.setInt(count, (Integer) valueEntry.getValue());
+			} else if (valueEntry.getValue() instanceof String) {
+				stmt.setString(count, (String) valueEntry.getValue());
+			} else if (valueEntry.getValue() instanceof Date) {
+				stmt.setDate(count, (Date) valueEntry.getValue());
+			} else if (valueEntry.getValue() instanceof Double) {
+				stmt.setDouble(count, (Double) valueEntry.getValue());
+			} else {
+				// handle for null
+				stmt.setString(count, null);
+			}
+			count++;
+		}
+	}
+
+	@Override
+	protected UserEntity getEntityFromResultSet(ResultSet resultSet)
+			throws SQLException, ClassNotFoundException, IOException {
+		BufferedImage img = null;
+		String encodedImg = null;
```

Comment:
Don't need to init img and encodedImg here like this. Just put it where they are used